ibmcloud/README.md

```diff
@@ -0,0 +1,269 @@
+As usual, you need to create `terraform.tfvars` to specify parameter values. `terraform.tfvars` looks like this.
+
+```
+ibmcloud_api_key = "<your API key>"
+ssh_key_name = "<your SSH key name>"
+cluster_name = "<cluster name>"
+```
+
+`ssh_key_name` is a name of your SSH key registered in IBM Cloud.
+You can add your SSH key at [https://cloud.ibm.com/vpc-ext/compute/sshKeys](https://cloud.ibm.com/vpc-ext/compute/sshKeys). This ssh key will be installed on control-plane and worker nodes.
+
+`cluster_name` is a name of a Kubernetes cluster. This name is used for the prefix of the names of control-plane and worker nodes. If you want to create another cluster in the same VPC, you need to use a different name for the new cluster.
+
+Then, execute the following commands to create a new Kubernetes cluster consisting of two virtual server instances. One for a control-plane node, and another one for a worker node. Please check [main.tf](terraform/cluster/main.tf) for the details.
+
+```
+cd ibmcloud/terraform/cluster
+terraform init
+terraform plan
+terraform apply
+```
+
+You can check the status of provisioned Kubernetes node VM instances at [https://cloud.ibm.com/vpc-ext/compute/vs](https://cloud.ibm.com/vpc-ext/compute/vs).
+
+This Terraform configuration also triggers execution of an Ansible playbook to set up
+Kubernetes and other prerequisite software in the two nodes. Please check [ansible/playbook.yml](terraform/cluster/ansible/playbook.yml) for the details.
+
+If ansible fails for some reason, you can rerun the ansible playbook as follows.
+```
+cd ansible
+ansible-playbook -i ./inventory -u root ./playbook.yml
+```
+
+When ansible fails, Terraform does not execute the setup script for Kubernetes. In this case, you can manually run it as follows. Note that you do not need to run this script manually, when everything goes well.
+
+```
+./scripts/setup.sh --bastion <floating IP of the worker node> --control-plane <IP address of the control-plane node> --workers  <IP address of the worker node>
+```
+
+When two VSIs are successfully provisioned, a floating IP address is assigned to the worker node. Please use the floating IP address to access the worker node from the Internet.
+
+## Build a pod VM image
+
+You need to build a pod VM image for peer pod VMs. A pod VM image contains the following components.
+
+* Kata agent
+* Agent protocol forwarder
+* skopeo
+* umoci
+
+The build scripts are located in [ibmcloud/image](./image). The prerequisite software to build a pod VM image is already installed in the worker node by [the Ansible playbook](terraform/cluster/ansible/playbook.yml) for convenience. Note that building a pod VM image on a worker node is not recommended for production, and we need to build a pod VM image somewhere secure to protect workloads running in a peer pod VM.
+
+The following command builds a custom VM image. A new QCOW2 file with prefix `podvm-` will be created in the current directory.
+
+```
+cd ibmcloud/image
+make build
```

Comment:
Assuming that these instructions code should run on the 'build machine' rather than on a VSI instance, it's probably worth adding `make` as a pre-req.